enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -280,15 +280,17 @@ object SlotAllocation {
            _: ConditionalApply =>
         rhsPipeline
 
-      case _: CartesianProduct =>
-        val newPipeline = lhsPipeline.breakPipelineAndClone()
+      case _: CartesianProduct |
+           _: NodeHashJoin =>
```

Comment:
Is this sneaky addition of `NodeHashJoin` intentional? Wondering about tests etc?

Reply:
Oops. Not by design!